pandas/core/window.py

```diff
@@ -341,6 +353,138 @@ def _center_window(self, result, window) -> np.ndarray:
                 result = np.copy(result[tuple(lead_indexer)])
         return result
 
+    def _get_roll_func(
+        self, cfunc: Callable, check_minp: Callable, index: np.ndarray, **kwargs
+    ) -> Callable:
+        """
+        Wrap rolling function to check values passed.
+
+        Parameters
+        ----------
+        cfunc : callable
+            Cython function used to calculate rolling statistics
+        check_minp : callable
+            function to check minimum period parameter
+        index : ndarray
+            used for variable window
+
+        Returns
+        -------
+        func : callable
+        """
+
+        def func(arg, window, min_periods=None, closed=None):
+            minp = check_minp(min_periods, window)
+            return cfunc(arg, window, minp, index, closed, **kwargs)
+
+        return func
+
+    def _apply(
+        self,
+        func: Union[str, Callable],
+        name: Optional[str] = None,
+        window: Optional[Union[int, str]] = None,
+        center: Optional[bool] = None,
+        check_minp: Optional[Callable] = None,
+        **kwargs
+    ) -> FrameOrSeries:
+        """
+        Rolling statistical measure using supplied function.
+
+        Designed to be used with passed-in Cython array-based functions.
+
+        Parameters
+        ----------
+        func : str/callable to apply
+        name : str, optional
+           name of this function
+        window : int/str, default to _get_window()
+            window lenght or offset
+        center : bool, default to self.center
+        check_minp : function, default to _use_window
+        **kwargs
+            additional arguments for rolling function and window function
+
+        Returns
+        -------
+        y : type of input
+        """
+        if center is None:
+            center = self.center
+
+        if check_minp is None:
+            check_minp = _use_window
+
+        if window is None:
+            window = self._get_window(**kwargs)
+
+        blocks, obj = self._create_blocks()
+        block_list = list(blocks)
+        index_as_array = self._get_index()
+
+        results = []
+        exclude = []
```

Comment:
> pandas/core/window.py:426: error: Need type annotation for 'exclude'

not sure what confuses mypy to require type for exclude (this is the reason ci fails)